source/code/source/playfab/PlayFabWinHttpPlugin.cpp

```diff
@@ -200,7 +200,11 @@ namespace PlayFab
                                 if (obj.first.length() != 0 && obj.second.length() != 0) // no empty keys or values in headers
                                 {
                                     std::string header = obj.first + ": " + obj.second;
-                                    WinHttpAddRequestHeaders(hRequest, std::wstring(header.begin(), header.end()).c_str(), -1, 0);
+                                    HRESULT hHeaderAdd = WinHttpAddRequestHeaders(hRequest, std::wstring(header.begin(), header.end()).c_str(), -1, 0);
+                                    if(!hHeaderAdd)
+                                    {
+                                        SetErrorInfo(reqContainer, "Error in WinHttpAddRequestHeaders");
```

Comment:
this will set the error info, but won't it still fire off the request? won't the response from that request overwrite the error info? it doesn't look like anything here is actually failing this loop early

Reply:
I can add a break after the SetErrorInfo to cut out early

Reply:
I'll leave it to you to fix it, but I think just adding a break after seterrorinfo will only break out of this "add headers" loop which still won't stop the http request from being sent off

Reply:
Yeah, I noticed the break would only break out one loop. I opted to instead do a return from the function, and created a helper that would close any open connections we may have made (moved the end part of the ExecuteRequest function to this new helper function and call it in 3 places, first 2 will set an error before hand and then return right after, and the final one is the correct happy path to complete the ExecuteRequest.